flask_appbuilder/security/sqla/apis/user/user.py

```diff
@@ -0,0 +1,200 @@
+from sqlalchemy.exc import IntegrityError
+from werkzeug.security import generate_password_hash
+
+from .schema import UserPostSchema, UserPutSchema
+
+
+class UserApi(ModelRestApi):
+    resource_name = "user"
+    class_permission_name = "User"
+    datamodel = SQLAInterface(User)
+    allow_browser_login = True
+
+    list_columns = [
+        "id",
+        "roles",
+        "first_name",
+        "last_name",
+        "username",
+        "active",
+        "email",
+        "last_login",
+        "login_count",
+        "fail_login_count",
+        "created_on",
+        "changed_on",
+        "created_by.id",
+        "changed_by.id",
+        "created.id",
+        "changed.id",
+    ]
+    show_columns = list_columns
+    add_columns = [
+        "roles",
+        "first_name",
+        "last_name",
+        "username",
+        "active",
+        "email",
+        "password",
+    ]
+    edit_columns = add_columns
+    search_columns = list_columns
+
+    add_model_schema = UserPostSchema()
+    edit_model_schema = UserPutSchema()
+
+    def pre_update(self, item):
+        item.changed_on = datetime.now()
+        item.changed_by_fk = g.user.id
+        if item.password:
+            item.password = generate_password_hash(item.password)
+
+    def pre_add(self, item):
+        item.password = generate_password_hash(item.password)
+
+    @expose("/", methods=["POST"])
+    def post(self):
+        """Create new user
+        ---
+        post:
+          requestBody:
+            description: Model schema
+            required: true
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+          responses:
+            201:
+              description: Item changed
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      result:
+                        $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        try:
+            item = self.add_model_schema.load(request.json)
+            model = User()
+            roles = []
+            for key, value in item.items():
+                if key != "roles":
+                    setattr(model, key, value)
+                else:
+                    for role_id in item[key]:
+                        role = (
+                            current_app.appbuilder.get_session.query(Role)
+                            .filter(Role.id == role_id)
+                            .first()
+                        )
```

Comment:
if role is None?

Reply:
actually there is a validator in schema, to check length at least 1 and required true so I assumed it won't be null or empty list for POST /user.

Reply:
handled and added test.